src/main/java/org/testng/TestRunner.java

```diff
@@ -968,6 +980,14 @@ private TestMethodWorker createTestMethodWorker(
     return vResult;
   }
 
+  private <T> T[] listToArray(List<T> sl, Class<T> clazz) {
```

Comment:
The method is not likely needed at all and can be replaced by List's `toArray(T[] a)` method